test/e2e/log_collector.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/pkg/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kinderrors "sigs.k8s.io/kind/pkg/errors"
+)
+
+const (
+	VSpherePrivateKey = "VSPHERE_SSH_PRIVATE_KEY"
```

Comment:
from looking at the e2e artifacts directory, it seems like we're collecting stuff (e.g. cloud-init-output.log etc..). which ssh key are we using ?

Reply:
The idea is to add a new ssh key to the prow job which is used to run the tests.

Reply:
@srm09 - how are we currently getting [this](https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-vsphere/1110/pull-cluster-api-provider-vsphere-e2e/1359984816526725120/artifacts/clusters/quick-start-y2vn14/machines/quick-start-y2vn14-rt598/) without an ssh key ?

Reply:
They're all empty files. The changes create the file before attempting to execute the command